core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -143,7 +201,24 @@ private void cacheManifests(FileIO fileIO) {
 
     if (allManifests == null) {
       // if manifests isn't set, then the snapshotFile is set and should be read to get the list
-      this.allManifests = ManifestLists.read(fileIO.newInputFile(manifestListLocation));
+      InputFile manifestListInputFile = fileIO.newInputFile(manifestListLocation);
```

Comment:
don't call if encrypted

Reply:
Why not add a call to `EncryptingFileIO` to open a `ManifestListFile` so that we can do the same thing here that we do for `ManifestFile`?

Reply:
I think that the change I suggested above is needed to avoid decryption in `SnapshotParser`. The parser should create a `ManifestListFile` with the encrypted key metadata and wrapped encryption key. Then when the manifest list is read by this `fileIO`, the key metadata is decrypted and passed in to create the decrypting stream.